### ISSUE TITLE
Update DeCLUTR requirements?

### ISSUE THREAD

**Issue by piegu** [JohnGiorgi/DeCLUTR#253]
Hi,

Actual DeCLUTR needs Pytorch 1.6 because of the allennlp version used. 
The actual version of Pytorch is 1.11 (March 2022).

Any plans to update DeCLUTR requirements?

Thanks.

**Comment by JohnGiorgi** [comment]
Hi,

Is this necessary? The [`setup.py` pins `allennlp` to `>=1.1.0, <1.2.0`](https://github.com/JohnGiorgi/DeCLUTR/blob/master/setup.py#L44) which itself depends on `torch`, so installing DeCLUTR (via `pip install git+https://github.com/JohnGiorgi/DeCLUTR.git`) _should_ give you the correct version of `torch`.

I just confirmed by running `pip install git+https://github.com/JohnGiorgi/DeCLUTR.git` in one of our [example notebooks](https://colab.research.google.com/github/JohnGiorgi/DeCLUTR/blob/master/notebooks/training.ipynb) and that is indeed the case. See the truncated output below:

```
Collecting git+https://github.com/JohnGiorgi/DeCLUTR.git
  Cloning https://github.com/JohnGiorgi/DeCLUTR.git to /tmp/pip-req-build-19yvqrqh
  Running command git clone -q https://github.com/JohnGiorgi/DeCLUTR.git /tmp/pip-req-build-19yvqrqh
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting allennlp<1.2.0,>=1.1.0
  Downloading allennlp-1.1.0-py3-none-any.whl (485 kB)
     |████████████████████████████████| 485 kB 4.3 MB/s 
Collecting torch<1.7.0,>=1.6.0
  Downloading torch-1.6.0-cp37-cp37m-manylinux1_x86_64.whl (748.8 MB)
     |████████████████████████████████| 748.8 MB 19 kB/s 
```

Are you having issues installing?

**Comment by piegu** [comment]
Hi @JohnGiorgi,

Thanks for your response (and for DeCLUTR!).
My problem is that I am doing a university research project that needs training DeCLUTR on a specific dataset.

Our great chance is that the university has a DGX A100 de NVIDIA (lucky we are) but... it appears that DGX A100 does not accept Pytorch 1.6

Then, with the current requirements, script and notebook training.ipynb, I can not train DeCLUTR.

Do you know how I could train it on DGX A100?

Thanks.

**Comment by JohnGiorgi** [comment]
Can you try updating torch to the version you need after installing DeCLUTR?

**Comment by piegu** [comment]
Hi @JohnGiorgi.

I did solve my problem thanks to a colleague:

1. He installed the [Pytorch release 20.07](https://docs.nvidia.com/deeplearning/frameworks/pytorch-release-notes/rel_20-07.html#rel_20-07) with Pytorch 1.6 (release that fits with DGX A100 de NVIDIA).
2. Then, he tried to install the DeCLUTR pip, but he had a problem, when installing a pytorch-metric-learning dependency, which the DeCLUTR repository requires, it kept trying to install the torch 1.10 cpu version of the pip library. This is not ideal, because the pytorch installation that comes in the container image is already optimized and built specifically to work with the versions of the CUDA drivers installed. He did not want to install another version of pytorch. So, he downloaded locally the "pytorch-metric-learning" and "allennlp" libraries repository and removed the pytorch dependency from within their setup.py. So forcing pip to install all the necessary dependencies except the torch that is already installed by the container image.

And it worked!
